### PR TITLE
Project Setup 

### PR DESCRIPTION
**Pull request by SriramPatibanda**
# Your checklist for this pull request

🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your default branch!
- [x] Make sure you are making a pull request against the **default branch** (left side). Also you should start *your branch* off *default branch*.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.
- [x] I have added necessary documentation (if appropriate)

## Description
- Setup the project using bloc
- Authentication using Firebase

### Work to be completed:
- Routing
- Modularizing Sign in page UI.
- Bloc Provider usage needs to be improved.

💖Thank you!

## Post merge checklist

- [ ] Follow steps from the [guidelines for contributing](../CONTRIBUTING.md) to this repository.
- [ ] If you are a new contributor, ping in the thread and one of the maintainers will add you to all-contributors list.

### CODE REVIEW

**Review by nayakastha — DISMISSED** [review]

File: elaichi/lib/app/theme/base_theme.dart [hunk 1]
```diff
@@ -0,0 +1,119 @@
+// ignore_for_file: avoid_classes_with_only_static_members
+import 'package:elaichi/app/theme/colors.dart';
+import 'package:flutter/material.dart';
+import 'package:google_fonts/google_fonts.dart';
```
Comment (abhibhaw) [inline elaichi/lib/app/theme/base_theme.dart:4]: Any specific reason to use a font package when we can add it manually? Will it increase the overall pub size? 🤨

File: elaichi/lib/app/theme/base_theme.dart [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    ),
+    colorScheme: const ColorScheme.dark(
+      primary: _lightOnPrimaryColor,
+      surface: Color.fromRGBO(27, 27, 27, 1),
+      primaryVariant: _lightOnPrimaryColor,
+      secondary: _lightPrimaryColor,
+      onPrimary: _lightPrimaryColor,
+    ),
+    floatingActionButtonTheme: const FloatingActionButtonThemeData(
+      backgroundColor: Colors.white,
```
Comment (abhibhaw) [inline elaichi/lib/app/theme/base_theme.dart:115]: why these colors are not imported through common file

File: elaichi/lib/auth/presentation/view/sign_in_page.dart [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  /// Constructor for [SignInPage]
+  const SignInPage({
+    Key? key,
+    required AuthenticationRepository authenticationRepository,
+  })  : _authenticationRepository = authenticationRepository,
+        super(key: key);
+
+  static const String _avenueLogoFileUri = 'assets/images/avenue_logo.png';
+  static const String _plantLogoFileUri = 'assets/images/plants.png';
+  static const String _googleLogoFileUri = 'assets/images/google_logo.svg';
```
Comment (abhibhaw) [inline elaichi/lib/auth/presentation/view/sign_in_page.dart:19]: why not even having these in a separate file 😄

File: elaichi/lib/home/view/home_page.dart [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+/// The Landing Page of the app.
+class HomePage extends StatelessWidget {
+  /// Constructor for [HomePage].
+  const HomePage({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return const Scaffold(
+      body: Center(
+        child: Text('HomePage'),
```
Comment (abhibhaw) [inline elaichi/lib/home/view/home_page.dart:12]: Please add a log out button, it's very essential for testing 🗡️

**Review by abhibhaw — CHANGES_REQUESTED** [review]

File: elaichi/lib/auth/presentation/view/sign_in_page.dart (hunk 3 above)
Comment (SriramPatibanda) [reply elaichi/lib/auth/presentation/view/sign_in_page.dart]: Done

File: elaichi/lib/auth/presentation/view/sign_in_page.dart [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+  final AuthenticationRepository _authenticationRepository;
+
+  @override
+  Widget build(BuildContext context) {
+    return RepositoryProvider<AuthenticationRepository>.value(
+      value: _authenticationRepository,
+      child: BlocProvider<SignInCubit>(
+        create: (context) => SignInCubit(_authenticationRepository),
+        child: SafeArea(
```
Comment (abhibhaw) [inline elaichi/lib/auth/presentation/view/sign_in_page.dart:27]: Any reason to use 2 scaffold

**Review by abhibhaw — APPROVED** [review]
lgtm

File: elaichi/lib/widgets/buttons.dart [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+          ),
+          padding: const EdgeInsets.all(16),
+        ),
+        onPressed: onTapped as void Function(),
+        child: Row(
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            icon!,
+            const Padding(
+              padding: EdgeInsets.only(left: 6),
```
Comment (nayakastha) [inline elaichi/lib/widgets/buttons.dart:40]: Are we not using relative dimensions?

**Review by nayakastha — APPROVED** [review]

**Pull request merged**